src/namepath.rs

```diff
@@ -1,8 +1,21 @@
 use super::*;
 
-#[derive(Clone, Debug, Eq, PartialEq, Ord, PartialOrd)]
+#[derive(Clone, Debug, Eq, Ord, PartialOrd)]
 pub(crate) struct Namepath<'src>(Vec<Name<'src>>);
 
+impl PartialEq for Namepath<'_> {
```

Comment:
I don't think we should implement a `PartialEq` implementation which ignores everything but the lexeme.

I think we should do `fn is_same_path(&self, other: &Self) -> bool` which compares the lexemes.